venues/ICLR.cc/2018/Conference/python/assign-reviewer.py

```diff
@@ -89,26 +142,45 @@ def next_anonreviewer_id(empty_anonreviewer_groups):
     client.remove_members_from_group(reviewers_group, reviewer_to_remove)
 
 if reviewer_to_add:
-    anonreviewer_id = next_anonreviewer_id(empty_anonreviewer_groups)
-    paper_authors = 'ICLR.cc/2018/Conference/Paper{0}/Authors'.format(paper_number)
-
-    anonymous_reviewer_group = openreview.Group(
-        id = anonreviewer_id,
-        readers = [
-            'ICLR.cc/2018/Conference',
-            'ICLR.cc/2018/Conference/Area_Chairs',
-            'ICLR.cc/2018/Conference/Program_Chairs',
-            anonreviewer_id
-            ],
-        nonreaders = [
-            paper_authors
-            ],
-        writers = ['ICLR.cc/2018/Conference'],
-        signatories = [anonreviewer_id],
-        signatures = ['ICLR.cc/2018/Conference'],
-        members = [reviewer_to_add])
-
-    print "adding {0} to {1}".format(reviewer_to_add, anonymous_reviewer_group.id)
-    client.post_group(anonymous_reviewer_group)
-    print "adding {0} to {1}".format(reviewer_to_add, reviewers_group.id)
-    client.add_members_to_group(reviewers_group, reviewer_to_add)
+    paper = client.get_notes(invitation='ICLR.cc/2018/Conference/-/Submission', number=paper_number)[0]
+    user_domain_conflicts, user_relation_conflicts = get_user_conflicts(reviewer_to_add)
+    paper_domain_conflicts, paper_relation_conflicts = get_paper_conflicts(paper)
```

Comment:
why dont you get the user conflicts here? iterate over all the authors and get the profile, if the profile doesn't exist then user the email domain.